### ISSUE TITLE
Error when running in container clusters

### ISSUE THREAD

**Issue by zer0def** [cgmartin/envhandlebars#7]
When an application uses this library to handle environment variables and is ran in a container cluster (like Marathon or Kubernetes), where envvars set by such software could be `PORT=<container_port>` and `PORT_<lb_port>=<container_port>`, for the example variables of `PORT=2345` and `PORT_1234=2345` a following backtrace is thrown and the application errors out.

```
/usr/local/lib/node_modules/envhandlebars/index.js:74
                vars[match[1]][match[2]] = opts.env[key];
                                         ^
TypeError: Cannot create property '1234' on string '2345'
    at parseVar (/usr/local/lib/node_modules/envhandlebars/index.js:74:42)
    at /usr/local/lib/node_modules/envhandlebars/index.js:52:13
    at Array.forEach (<anonymous>)
    at getVars (/usr/local/lib/node_modules/envhandlebars/index.js:51:31)
    at applyTemplate (/usr/local/lib/node_modules/envhandlebars/index.js:40:35)
    at ConcatStream.<anonymous> (/usr/local/lib/node_modules/envhandlebars/node_modules/concat-stream/index.js:36:43)
    at ConcatStream.emit (events.js:164:20)
    at finishMaybe (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:607:14)
    at endWritable (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:615:3)
    at ConcatStream.Writable.end (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:571:41)
Converting: /opt/admin/env/cfg.json.tpl
/usr/local/lib/node_modules/envhandlebars/index.js:74
                vars[match[1]][match[2]] = opts.env[key];
                                         ^
TypeError: Cannot create property '1234' on string '2345'
    at parseVar (/usr/local/lib/node_modules/envhandlebars/index.js:74:42)
    at /usr/local/lib/node_modules/envhandlebars/index.js:52:13
    at Array.forEach (<anonymous>)
    at getVars (/usr/local/lib/node_modules/envhandlebars/index.js:51:31)
    at applyTemplate (/usr/local/lib/node_modules/envhandlebars/index.js:40:35)
    at ConcatStream.<anonymous> (/usr/local/lib/node_modules/envhandlebars/node_modules/concat-stream/index.js:36:43)
    at ConcatStream.emit (events.js:164:20)
    at finishMaybe (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:607:14)
    at endWritable (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:615:3)
    at ConcatStream.Writable.end (/usr/local/lib/node_modules/envhandlebars/node_modules/readable-stream/lib/_stream_writable.js:571:41)
```

**Comment by luisdavim** [comment]
#8 should fix this:
```
➜ 18:47:26 ~/.../PTC-Global/envhandlebars (master|…1) $ export PORT_1234=2345
➜ 18:47:41 ~/.../PTC-Global/envhandlebars (master|…1) $ echo "The port is {{PORT_1234}}" | ./bin/envhandlebars --no_arrays
The port is 2345
➜ 18:47:43 ~/.../PTC-Global/envhandlebars (master|…1) $ echo "The port is {{PORT_1234}}" | ./bin/envhandlebars
The port is
```

**Comment by cgmartin** [comment]
@luisdavim regarding the #8 `--no_arrays` option, what do you think about the following alternative solution:

New argument `--array_var_prefix=FOO_` which would explicitly process only those vars with the prefix (ie. `FOO_NAME_1`) for the iterator and array-of-object features. All other vars without the prefix would be skipped. This could allow the uncontrolled/3rd-party vars to coexist with user-controlled vars and the array features to still be used.

**Comment by luisdavim** [comment]
That's a great idea. I'll open a PR for that.

**Comment by luisdavim** [comment]
Hi, did you get a change to get this released?

**Comment by cgmartin** [comment]
The new argument features have now been released in `envhandlebars@1.4.0`. There were some minor changes in #10 to the argument option names. Thanks again @luisdavim for your contributions!

**Comment by luisdavim** [comment]
Awesome, thanks.
